src/ore/src/task.rs

```diff
@@ -74,72 +76,114 @@ impl<T> Future for AbortOnDropHandle<T> {
     }
 }
 
-/// Extension methods for [`JoinHandle`].
-#[async_trait::async_trait]
-pub trait JoinHandleExt<T> {
-    /// Converts a [`JoinHandle`] into a [`AbortOnDropHandle`].
-    fn abort_on_drop(self) -> AbortOnDropHandle<T>;
+/// Wraps a tokio `JoinHandle` and provides 4 exclusive (i.e. they take `self` ownership)
+/// operations:
+///
+/// - `abort_on_drop`: create an `AbortOnDropHandle`.
```

Comment:
```suggestion
/// - `abort_on_drop`: create an `AbortOnDropHandle` that will automatically abort the task
///   when the handle is dropped.
```

Seems nice to spell this out!